update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+            legacy_suffix += to_base26(vo_id)
+            if legacy_suffix not in account_suffix:
+                is_unique = True
+            j += 1
+    return legacy_suffix
+
+
+def get_account_suffix(vo_name, fqan):
+    """Returns either a predefined suffix for pilot, production and software
+       manager roles, or a 3-character suffix corresponding to the base26-like
+       encoding of the FQAN hashcode
+    """
+    if fqan['type'] in FQAN_TYPE_MAPPING:
+        return FQAN_TYPE_MAPPING[fqan['type']]
+    else:
+        relative_fqan = fqan['expr'].replace('/' + vo_name, '', 1)
+        return to_base26(java_string_hashcode(relative_fqan))
+
+
+def write_cert_info(voms_endpoint):
+    """Writes subject and issuer of VOMS server valid certificates as a
+       dict element
+    """
+    content = ""
+    subject = ""
+    issuer = ""
+    if voms_endpoint.cert:
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, voms_endpoint.cert)
+        for (field, value) in cert.get_subject().get_components():
+            subject = subject + "/" + field + "=" + value
+        for (field, value) in cert.get_issuer().get_components():
+            issuer = issuer + "/" + field + "=" + value
+        content = format("    '%s', dict(\n" % voms_endpoint.hostname)
+        content += format("        'subject', '%s',\n" % subject)
+        content += format("        'issuer', '%s',\n" % issuer)
+        content += format("    ),\n")
+    return content
+
+
+def write_vo_template(vo_details, basedir, namespaces):
+    """Write VO template
+    """
+    vo_params_ns = namespaces['params_dir'] + os.path.sep + vo_details['name']
+    vo_params_tpl = basedir + os.path.sep + vo_params_ns + ".pan"
+    force_voms_admin = False
+
+    sys.stdout.write("Writing template for VO " + vo_details['name'] +
+                     " (" + vo_params_tpl + ")\n")
+    template = open(vo_params_tpl, 'w')
+    template.write("structure template " + vo_params_ns + ";\n\n")
+    template.write("'name' ?= '" + vo_details['name'] + "';\n")
+    vo_name = re.sub(re.compile('^vo\.'), '', vo_details['name'])
+    account_prefix = vo_name[0:3] + to_base26(vo_details['id'])
+    template.write("'account_prefix' ?= '" + account_prefix + "';\n\n")
+
+    # VOMS Servers
+    template.write("'voms_servers' ?= list(\n")
+    if not vo_details['voms_endpoints']:
+        sys.stderr.write("WARNING: VO " + vo_details['name'] +
+                         " has no VOMS endpoint defined\n")
+    elif len(vo_details['voms_endpoints']) == 1:
+        force_voms_admin = True
+    for endpoint in vo_details['voms_endpoints']:
+        template.write("    dict(\n")
+        template.write("        'name', '%s',\n" % endpoint.hostname)
+        template.write("        'host', '%s',\n" % endpoint.hostname)
+        template.write("        'port', %s,\n" % endpoint.vomses_port)
+        template.write("        'adminport', %s,\n" % endpoint.https_port)
```

Comment:
Same for this template string, even better as this is being passed a dict, you can use the properties in the pattern:

```python
VOMS_ENDPOINT_TEMPLATE = '''
    dict(
        'name', '%(hostname)s',
        'host', '%(hostname)s',
        'port', %(vomses_port)s,
        'adminport', %(https_port)s,
'''
```
```python
    for endpoint in vo_details['voms_endpoints']:
        template.write(VOMS_ENDPOINT_TEMPLATE % endpoint)
```

Reply:
The syntax has been modified.